### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by tremby** [jellyfin/jellyfin-mpv-shim#258]
**Describe the bug**
I'm on a Raspberry Pi with 2G of ram.

Several times over the last few weeks I've had the system almost entirely lock up due to jellyfin-mpv-shim exhausting all its memory. I try to shell in and have to wait minutes before I can get a prompt so that I can check `htop` and kill the shim, then restart it.

I saw that this is mentioned in the readme. Prompted by that, I set the shim to use external mpv so I could verify what was eating the memory, and it's certainly mpv. The shim has stayed almost entirely fixed at around 26M usage. mpv starts low. When playing the first video I was seeing about 350M usage by mpv. After stopping it I see the usage go down very slightly, 10 or 20M. When starting another video it shoots up significantly; I saw 519M. Switched video again, it's up to 665M. Switched again, up to 873M after a few seconds. Stopped playback, it's down to 848M. Played exactly the same video again, and it's up to 1037M within a few seconds.

It seems to me that this is likely a bug with mpv rather than with jellyfin-mpv-shim, but it could be that the shim's usage of mpv is to blame, and since you asked in the readme for info, I'm posting here first.

**To Reproduce**
1. Switch on external mpv
2. Monitor memory usage
3. Play various videos, no need to let them play through, just flick through a few and see the memory usage of mpv rise
4. See the shim's memory stay pretty much constant
5. See that if you stop a video, memory usage of mpv goes down slightly, but remains much too high

**Expected behavior**
Maybe mpv would use different amounts of memory for each video, but after stopping it should free up pretty much everything. Memory usage should not keep climbing when playing multiple videos.

**Screenshots**
n/a

**Desktop (please complete the following information):**
 - OS: Raspberry Pi OS 10
 - Shim version: 2.0.2

**Error Messages**
Nothing interesting in the log.

**Comment by iwalton3** [comment]
Yes I’m aware of this issue. The enhancement to fix it without fixing MPV would be to kill MPV while it is not in use. But right now the player code expects MPV to be running at all times and that refactor will be kind of a pain to fix.

**Comment by tremby** [comment]
Yes, as I said I saw it mentioned on the readme, where you also say to let you know if it's MPV or the shim. That's what I'm doing here.

Are the MPV developers aware of it, to your knowledge?

If they're not going to fix it, a workaround kind of has to happen here in the shim. Can't have it just eating all the system memory when it's been running for a long period of time, since that's exactly the expected usage pattern.

**Comment by iwalton3** [comment]
I’m not sure. There are some old issues. I don’t usually see awful memory leak issues on my system anymore since I stopped using SVP. But it still isn’t ideal.

I do know that the RPI lacks hardware acceleration in MPV so it probably isn’t that…

**Comment by tremby** [comment]
Ah, that'll be why it struggles horribly with some higher-quality content, I guess.

(I really didn't want to go back to Kodi...)
